main.go

```diff
@@ -245,41 +256,66 @@ func main() {
-						possibleUsername := strings.TrimRight(strings.ToLower(strings.TrimLeft(field, prefix)), ",.? ")
-						if _, ok := usernames[possibleUsername]; ok {
-							log.Printf("Found username, adding to notifications %s!\n", possibleUsername)
-							pinger := c.Users[event.UserID].Nick
-							inOneHour := time.Now().In(location).Add(5 * time.Second)
-							notifications[usernames[possibleUsername]][messageID] = Notification{inOneHour, messageID, event.Flow, pinger}
-							notifyTag := fmt.Sprintf("notify-short-%v", possibleUsername)
-							flowdock.EditMessageInFlowWithApiKey(flowdockAPIKey, org, flow, strconv.FormatInt(event.ID, 10), "", []string{notifyTag})
-						}
-						log.Printf("%s", notifications[event.UserID][messageID])
-						break
+
+				var notifRegex = regexp.MustCompile(fmt.Sprintf(`(\%s+)([\wåäö]+)`, prefix))
+
+				for _, field := range notifRegex.FindAllStringSubmatch(event.Content.Text, -1) {
+					if len(field) < 2 {
+						continue
 					}
-					if strings.HasPrefix(field, prefix) && !strings.HasPrefix(field, strings.Repeat(prefix,2)) {
-						log.Printf("Saw notification thingie, %s", field)
-						possibleUsername := strings.TrimRight(strings.ToLower(strings.TrimLeft(field, prefix)), ",.? ")
-						if _, ok := usernames[possibleUsername]; ok {
-							log.Printf("Found username, adding to notifications %s!\n", possibleUsername)
-							pinger := c.Users[event.UserID].Nick
-							nextWorkDayAtNine := NextWorkdayAtNine()
-							//addNotification(nextWorkDayAtNine, possibleUsername, pinger, messageID)
-							notifications[usernames[possibleUsername]][messageID] = Notification{nextWorkDayAtNine, messageID, event.Flow, pinger}
-							notifyTag := fmt.Sprintf("notify-long-%v", possibleUsername)
-							flowdock.EditMessageInFlowWithApiKey(flowdockAPIKey, org, flow, strconv.FormatInt(event.ID, 10), "", []string{notifyTag})
-							saveNotifications(notifications, notificationStorage)
-						}
-						log.Printf("%s", notifications[event.UserID][messageID])
-						break
+					possiblePrefix := field[1]
+					possibleUsername := field[2]
+					// Check first if the username is a known username, if not skip
+					if _, ok := usernames[possibleUsername]; !ok {
+						continue
+					}
+					pinger := c.Users[event.UserID].Nick
+
+					var notificationTime time.Time
+					var notifyTag string
+					if possiblePrefix == fastPrefix {
+						notificationTime = time.Now().In(location).Add(fastDelay)
+						notifyTag = fmt.Sprintf("notify-short-%v", possibleUsername)
 					}
+					if possiblePrefix == slowPrefix {
+						notificationTime = NextWorkdayAtNine()
+						notifyTag = fmt.Sprintf("notify-long-%v", possibleUsername)
+					}
+					if possiblePrefix == fasterPrefix {
+						notificationTime = time.Now().In(location).Add(fasterDelay)
+						notifyTag = fmt.Sprintf("notify-shorter-%v", possibleUsername)
+					}
+					addNotification(notifications, notificationTime, possibleUsername, pinger, messageID, event.Flow, event.ID)
```

Comment:
As the code for messages and comments look quite identical, wouldn't it make sense to combine them to a common switch case or function?